portfolio/src/main/webapp/scripts/server-dev.js

```diff
@@ -15,15 +15,38 @@
 /**
  * Fetches the previously entered comments from the server and inserts each
  * comment as a list item of the 'comments' <ul> element.
+ *
+ * <p>
+ *
+ * An option to determine the maximum number of comments is also included
+ * using a Query String parameter created from the num-comments form. When
+ * the page is (re-)loaded, the number of comments displayed is determined
+ * from the selection in the previous session. Otherwise, the most recently
+ * submitted number selection will be used. The number of comments will
+ * also never exceed the number of total comments returned from the datastore.
  */
 function getCommentsThread() {
   fetch('/comment-data')
       .then(response => response.json())
       .then((commentList) => {
         const commentThread = document.getElementById('comments-thread');
-        commentList.forEach((comment) => {
-          commentThread.appendChild(createListElement(comment));
-        });
+        const urlParams = new URLSearchParams(window.location.search);
+
+        // Determine the number of comments to display
+        var numComments = urlParams.get('num-comments');
+        const numCommentsStored = parseInt(
```

Comment:
Is it possible for numCommentsStored to be less than commentList length?

Reply:
Yes for my implementation. What I was thinking is that the user can choose how many comments they want to see (capped it at 100 for now). Thus if the stored number is smaller than the full length of the list, that is okay. But, if `numCommentsStored` or 'numComments' from the URL param s larger than `commentsList.length`, all of the comments will be displayed. 

Please let me know if I did not understand your question here or if you recommend that there needs to be a change made here. Thanks!

Reply:
I think there's some confusion on the difference between commentsList.length and numCommentsStored. My impression was that numComments is the user-provided value (capped at 100?), numCommentsStored is the actual number of comments stored in the database, and commentList.length is the number of actual comments fetched. Is that correct? If so, then shouldn't numCommentsStored always be equal to commentList.length (assuming all comments are always fetched)?

Reply:
Yes I apologize for the confusion. The main confusion point here is numCommentsStored. This value corresponds to the last user-provided value and it is stored in the session instance using JS so that it will persist with reloads. The purpose of including this was if the user said they wanted to display 10 comments, this makes sure that if they add a new comment to the list, upon reload of the page after grabbing the new comment from the database, 10 comments are still displayed rather than the default value (5 I think). This is because upon reload, the numComments in the urlParams is null because it is not set so the default value would have been used. 

The only time that the commentList.length comes into play is if the chosen value (either stored or user chosen on the current page) is larger than the total number of comments stored. In this case, it limits it to the max size of the comments returned from the database.

Does this make sense? If so then I think I will re-name some of these variables so that they reflect what I just said. If you have variable name suggestions other than the ones in my next commit, please let me know.